### PR TITLE
feat: Add `domain` support

### PR DESCRIPTION
**Pull request by gungoren**
## Description

Add `domain` and `domain_iam_role_name` support to `aws_rds_cluster`
## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by gungoren** [comment]
Waiting to https://github.com/hashicorp/terraform-provider-aws/pull/35753 to be released with `5.37`

File: examples/postgresql-kerberos/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# PostgreSQL Example
```
Comment (bryantbiggs) [inline examples/postgresql-kerberos/README.md:1]: I don't think we need to show this in an example - adding a directory service resource will take a looooooooooooooooooooong time to test/validate. For this change we can skip the example usage - @antonbabenko what are your thoughts

Comment (antonbabenko) [reply examples/postgresql-kerberos/README.md:1]: I agree, we don't need to add such a long example. The feature itself adds just 2 new arguments to the module.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: examples/postgresql-kerberos/README.md (hunk 1 above)
Comment (gungoren) [reply examples/postgresql-kerberos/README.md]: kerberos example removed from PR.

**Review by antonbabenko — APPROVED** [review]

**Review by bryantbiggs — APPROVED** [review]
thank you!

**Comment by antonbabenko** [comment]
This PR is included in version 9.1.0 :tada:

**Pull request merged**